### PR TITLE
libswiften: add head build

### PR DESCRIPTION
**Pull request by vitalyster**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by vitalyster** [comment]
Let me know what can I do for this strange error on my side

File: Formula/libswiften.rb [hunk 1]
```diff
@@ -16,6 +16,10 @@ class Libswiften < Formula
     sha256 "0cd2296d234b0c59bcd9dc5e0ebf78f7439ac7c91e415efad693976d01666338" => :mavericks
   end
 
+  head do
```
Comment (dunn) [inline Formula/libswiften.rb]: This can be a single line `head "git://swift.im/swift"` below the `revision`.

**Comment by dunn** [comment]
Not sure why this is happening only on El Capitan:
```
Error: Failed changing install name in /usr/local/Cellar/libswiften/3.0_1/bin/sluift
  from /usr/local/opt/boost/lib/libboost_program_options.dylib
    to @@HOMEBREW_PREFIX@@/opt/boost/lib/libboost_program_options.dylib
==> Bottling libswiften-3.0_1.el_capitan.bottle.1.tar.gz...
Error: Updated load commands do not fit in the header of /usr/local/Cellar/libswiften/3.0_1/bin/sluift. /usr/local/Cellar/libswiften/3.0_1/bin/sluift needs to be relinked, possibly with -headerpad or -headerpad_max_install_names

**Comment by vitalyster** [comment]
https://github.com/Homebrew/homebrew-core/issues/7852 maybe outdated boost bottled for elcapitan?

**Comment by dunn** [comment]
@brewtestbot test this please

**Comment by fxcoudert** [comment]
It's the infamous `Error: Updated load commands do not fit in the header of /usr/local/Cellar/libswiften/3.0_1/bin/sluift` relocation error :(

**Comment by ilovezfs** [comment]
This is now fixed by https://github.com/Homebrew/homebrew-core/pull/15414.

**Comment by ilovezfs** [comment]
@vitalyster Thanks for the pull request! Shipped!

**Pull request merged**